libs/providers/unleash/src/lib/unleash-provider.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  EvaluationContext,
+  Logger,
+  Provider,
+  JsonValue,
+  ResolutionDetails,
+  OpenFeatureEventEmitter,
+  ProviderFatalError,
+  ProviderEvents,
+  FlagNotFoundError,
+  TypeMismatchError,
+} from '@openfeature/server-sdk';
+import { Unleash, initialize, UnleashConfig, Context } from 'unleash-client';
+
+export class UnleashProvider implements Provider {
+  metadata = {
+    name: UnleashProvider.name,
+  };
+
+  public readonly events = new OpenFeatureEventEmitter();
+
+  private _logger?: Logger;
+
+  // config is the Unleash config provided to the provider
+  private _config?: UnleashConfig;
+
+  // client is the Unleash client reference
+  private _client?: Unleash;
```

Comment:
I may be confused but I'm not sure where this is actually being assigned, `initialize` seems to return a void promise.